src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStore.cs

```diff
@@ -79,17 +85,22 @@ public Task CreateFileFromStream(string path, Stream inputStream, bool overwrite
 
         public string MapPathToPublicUrl(string path)
         {
-            return _publicUrlBase.TrimEnd('/') + "/" + this.NormalizePath(path);
+            var publicUrl = new PathString(_publicUrlBase.TrimEnd('/') + "/" + this.NormalizePath(path));
```

Comment:
Throws with Azure

> Microsoft.AspNetCore.Http.PathString..ctor(String value)
The path in 'value' must start with '/'.

`_publicUrlBase` is a full http URI like `https://youraccount.blob.core.windows.net/media`